yt/testing.py

```diff
@@ -118,8 +119,12 @@ def cubicspline_python(
 
 
 def integrate_kernel(
-    kernelfunc: Callable[[float], float], b: float, hsml: float
-) -> float:
+    kernelfunc: Callable[
+        [float | npt.NDArray[np.floating]], float | npt.NDArray[np.floating]
+    ],
+    b: float | npt.NDArray[np.floating],
+    hsml: float | npt.NDArray[np.floating],
+) -> float | npt.NDArray[np.floating]:
```

Comment:
It's often much preferable to be strict on the return type. Here I don't see any reason not to be
```suggestion
) -> float:
```